### ISSUE TITLE
Feature requests

### ISSUE THREAD

**Issue by tehniemer** [akhilrex/podgrab#9]
First off, thanks for publishing this, at first glance it seems much faster than a dockerized instance of gPodder that I've been running with for a while. However, the former has a few niceties, and missing features, that hopefully wouldn't be too hard to add:

- Folder structure and file naming/tagging settings.
- Directory scanning, #8 or a "mark as downloaded" button if scanning directories isn't a trivial task.
- A "download all" button to download episodes that aren't already on the disk.
- An unplayed/played flag.
- A grid layout option for the home page.
- OPML export (thanks for adding import).

**Comment by akhilrex** [comment]
I have added the following features. Just pull the latest image and rebuilt the container.

- A "download all" button to download episodes that aren't already on the disk.
- A delete all to delete episodes of a podcast
- An unplayed/played flag.


Following are in development

- OPML export (thanks for adding import).
- A grid layout option for the home page.

**Comment by demslam** [comment]
Excellent!

**Comment by undaunt** [comment]
Camping on to the above list, any chance to pass through UID and GID along with time zone into the image for permissioning? I frequently pass through all 3 in linuxserver images as: TZ, PUID, and PGID.

Thank you very much.

**Comment by akhilrex** [comment]
Hi @undaunt The background job runs at the frequency set in the env file (default 4 hrs). Please check if all episodes are downloaded after that interval has passed or not. If it is working you can reduce it to something smaller in the env file (30 mins maybe). I had originally kept a high default to avoid conflicts/locks of same job running twice but now that's handled in the code.

**Comment by undaunt** [comment]
I'll check, thanks!

Since conflicts are now handled in code could this action converted to an on demand action?

**Comment by akhilrex** [comment]
I'll look into it.

Thanks and Regards

Akhil Gupta

**Comment by akhilrex** [comment]
> I'll check, thanks!
> 
> Since conflicts are now handled in code could this action converted to an on demand action?

Hi. This was fixed a couple of releases back. Though I would let you know.

**Comment by undaunt** [comment]
Thanks! Do you know what processes may or may not be running that would prevent this from triggering on demand? I added six shows to download all episodes earlier today and they are still queued.

**Comment by akhilrex** [comment]
> First off, thanks for publishing this, at first glance it seems much faster than a dockerized instance of gPodder that I've been running with for a while. However, the former has a few niceties, and missing features, that hopefully wouldn't be too hard to add:
> 
> * Folder structure and file naming/tagging settings.
> * Directory scanning, #8 or a "mark as downloaded" button if scanning directories isn't a trivial task.
> * A "download all" button to download episodes that aren't already on the disk.
> * An unplayed/played flag.
> * A grid layout option for the home page.
> * OPML export (thanks for adding import).

@tehniemer OMPL Export is live. Please pull the latest image and it should be there on the settings page.

**Comment by undaunt** [comment]
@akhilrex I had some downloads queued for over four days and they didn't download so I just did a fresh install today. They still don't appear to be running on demand. Anything in the logs I could look out for?

Thank you.

**Comment by akhilrex** [comment]
Try these steps

Rebuild the container with latest image. Backup existing downloaded files
for this podcast
Delete the podcast.
Put the downloaded files back again.
Add the podcast again.

Let me know it this persists.

In case bandwidth is not an issue, you can skip the baking up and restoring
files steps and simply delete and re add podcast.

These episodes are not really stuck. I made a few changes in the logic some
time back. These are most likely incorrect dB flags.

Thanks and Regards

Akhil Gupta

**Comment by undaunt** [comment]
Hi Akhil,

I already rebuilt the container today. These issues persisted again today until I checked off 'download new episodes automatically' as I indicated above. Can you verify this checkbox is somehow being required for downloading all episodes of a podcast on demand?

**Comment by akhilrex** [comment]
That's strange. Please share the url of the podcast that you are trying and i will check it.

**Comment by undaunt** [comment]
It could be literally any podcast I have in my system. I'm wondering if the number of podcasts I have and the need to ping all of them for a 200 response when trying to download just 1 of them may be causing an issue? I have about 39 podcasts added.

For example:

[Serial](https://feeds.simplecast.com/xl36XBC2)
[Making](http://feeds.feedburner.com/wbez-making)
[Lost Mountain Saga](https://feeds.simplecast.com/Viae7d6b)

As soon as I enable the 'automatically download new episodes to disk', then attempt to download all episodes of a different podcasts, **both** podcasts begin downloading.

If I then uncheck the option and add another podcast to the 'download all' queue, it sits there again.

I tested this both with, and without, the 'download episodes whenever a new podcast is added' enabled and it didn't seem to impact things. Only when 'automatically download new episodes to disk' is enabled will the podcasts download all episodes.

Again, this also doesn't impact single episode downloads, which always trigger as soon as I click them.

Unrelated but tangentially related, is there a reason all podcasts need to be queued (asking since I have 39 of them) for a 200 response before the one I just tried to download will begin downloading?

Thanks very much.
